src/SFA.DAS.Assessor.Functions/Domain/Print/DeliveryNotificationCommand.cs

```diff
@@ -64,11 +67,17 @@ private async Task ProcessFiles(IEnumerable<string> fileNames)
 
         private async Task ProcessFile(string fileName)
         {
-            var receipt = JsonConvert.DeserializeObject<DeliveryReceipt>(_fileTransferClient.DownloadFile($"{_sftpSettings.DeliveryNotificationDirectory}/{fileName}"));
+            var fileContent = _fileTransferClient.DownloadFile($"{_sftpSettings.DeliveryNotificationDirectory}/{fileName}");
+            var receipt = JsonConvert.DeserializeObject<DeliveryReceipt>(fileContent);
+
+            _validationService.Start(fileName, fileContent, _sftpSettings.ErrorDeliveryNotificationDirectory);
```

Comment:
If an error is thrown before the .End() is reached then I think this will not write the logs files - suggest you use a using(IDisposable) pattern to guarantee that the .End() is called

Reply:
Ok changed to IDisposable